app/services/one_off_redirect_fix_service.rb

```diff
@@ -0,0 +1,43 @@
+class OneOffRedirectFixService
+  def self.fix_redirects!
+    editions = Edition.where(document_type: "redirect").joins(:document).where.not("documents.locale = 'en'")
+    editions.each do |edition|
+      updated = update_redirects(edition)
+      Commands::V2::RepresentDownstream.new.call([edition.content_id], queue: DownstreamQueue::LOW_QUEUE) if updated
+    end
+  end
+
+  def self.update_redirects(edition)
+    return false if edition.redirects.blank?
+
+    if edition.redirects.count > 1
+      Rails.logger.info "[RedirectDepthFix] edition #{edition.id} has more then one redirect"
+      return false
+    end
+
+    next_base_path = edition.redirects.first[:destination]
+
+    while (next_edition = Edition.where(base_path: next_base_path).order(:updated_at).last)
+      break unless next_edition && next_edition.redirects.present?
+
+      if next_edition.redirects.count > 1
+        Rails.logger.info "[RedirectDepthFix] edition #{next_edition.id} has more then one redirect"
+        return false
+      end
+
+      next_base_path = next_edition.redirects.first[:destination]
+
+      if next_base_path == edition.base_path
+        Rails.logger.info "[RedirectDepthFix] edition #{edition.id} and #{next_edition.id} has circular dependencies"
+        return false
+      end
+    end
+
+    redirect = edition.redirects.first
+    redirect[:destination] = next_base_path
+    edition.redirects = [redirect]
```

Comment:
Can this be shortened to this?

```suggestion
    edition.redirects.first[:destination] = next_base_path
```

Reply:
If we shorten this code, the edition will not get updated, we have to update destination value first.